README.md

```diff
@@ -289,18 +289,40 @@ Just as with React, but assuming Angular instead :)
 
 As a bootstrap for the application we need something like:
 
+```typescript 
+@Injectable()
+export class TodoService {}
+
+@NgModule({
+  declarations: [
+    AppComponent,
+    TodoListComponent
+  ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    FormsModule
+  ],
+  providers: [
+    {provide: TodoService, useValue: Telescope.of<any>(TodosState.empty()) }
```

Comment:
Telescope<T> is not a subtype of Service and does not implement the interface. Consider one of the following alternatives:

- Change TodoService from a class to an interface and expose the telescope as a property.
- Change TodoService to an interface and implement the Telescope contract in addition to the application specific methods you consider relevant, in the example it can be represented with a single comment.
- Directly inject the telescope you need using a token and an instance provider.

Reply:
Awesome - thank you for the suggestions - I solved it by `injecting` `Telescope` into the service, 
and then using it inside the services, 

**I am a little torn, between** 
1. `injecting` `Telescope` into each service - this would allow for each service to manage their 
state - then however `lenses` would not be as useful - because you would only be magnifying specific `todos` 
https://github.com/georgebatalinski/telescope_todo/blob/master/src/app/todos/todo.service.ts#L15

2. Keeping global state - makes a lot of boiler plate :( on the upside you are using only 1 telescope
https://github.com/georgebatalinski/telescope_todo/blob/master/src/app/app.state.ts#L13


@alexeieleusis what do you think about this?

Reply:
I'm not an angular expert, but I feel that there are two conversations: One is how to have "local telescopes", if what you want is to have a "local" state on each component that somehow "includes" part of the global state:

`LocalState = (SomeProjectionOf GlobalState, StuffOnlyImportantForTheComponent)?`

If that is the case, one option is to magnify the global telescope T into a "smaller" one T'; inject or provide as input to a child component and there use `uplift` (which BTW needs some documentation in the README too) to create a new Telescope with a combined state. When the part of the state exclusive to the component is changed, the global telescope will not emit, but if the shared part do change, then the global stream will emit.

The second conversation about boilerplate is more specific to Angular and I prefer to let @alexeieleusis take that part.

Reply:
I will look at `uplift` and see if i can add some documentation on it - thank you for suggestion 😄

Reply:
Just added some notes on the Readme (still pending for review) for the uplift method, my comment in the code and the code itself probably were not completely self explanatory :(

Reply:
It looked really good - but I will re-over again